internal/driver/config.go

```diff
@@ -222,22 +240,34 @@ func createTcpConnectionInfo(tcpProtocol map[string]any) (info *ConnectionInfo,
 		return nil, fmt.Errorf("uintID value out of range(0–255). Error: %v", err)
 	}
 
-	timeout, err := parseIntValue(tcpProtocol, Timeout)
+	timeout, err := parseFloatValue(tcpProtocol, Timeout)
 	if err != nil {
 		return nil, err
 	}
 
-	idleTimeout, err := parseIntValue(tcpProtocol, IdleTimeout)
+	idleTimeout, err := parseFloatValue(tcpProtocol, IdleTimeout)
 	if err != nil {
 		return nil, err
 	}
+	var linkRecoveryTimeout *float64
+	linkRecoveryTimeoutFloatVal, err := parseFloatValue(tcpProtocol, LinkRecoveryTimeout)
+	if err == nil {
+		linkRecoveryTimeout = &linkRecoveryTimeoutFloatVal
+	}
+	var protocolRecoveryTimeout *float64
+	protocolRecoveryTimeoutFloatVal, err := parseFloatValue(tcpProtocol, ProtocolRecoveryTimeout)
+	if err == nil {
+		protocolRecoveryTimeout = &protocolRecoveryTimeoutFloatVal
+	}
```

Comment:
these recovery wouldn't work if no value is passed (mostly for backwards compatibility and ease of use). Should these have default values or are we good?

Reply:
The grid-x/modbus lib doesn't provide the default timeout for both of them, so we should not set the default timeout, either.
It's hard to say what default timeout is better for the general usage.